tools/distrib/python/grpcio_tools/grpc_tools/protoc.py

```diff
@@ -31,6 +36,124 @@ def main(command_arguments):
     return _protoc_compiler.run_main(command_arguments)
 
 
+if sys.version_info[0] > 2:
+    import contextlib
+    import importlib
+    import importlib.machinery
+    import threading
+
+    def _module_name_to_proto_file(suffix, module_name):
+        components = module_name.split(".")
+        proto_name = components[-1][:-1 * len(suffix)]
+        return os.path.sep.join(components[:-1] + [proto_name + ".proto"])
+
+    def _proto_file_to_module_name(suffix, proto_file):
+        components = proto_file.split(os.path.sep)
+        proto_base_name = os.path.splitext(components[-1])[0]
+        return ".".join(components[:-1] + [proto_base_name + suffix])
+
+    @contextlib.contextmanager
+    def _augmented_syspath(new_paths):
+        original_sys_path = sys.path
+        if new_paths is not None:
+            sys.path = sys.path + new_paths
+        try:
+            yield
+        finally:
+            sys.path = original_sys_path
+
+    # TODO: Investigate making this even more of a no-op in the case that we have
+    # truly already imported the module.
+    def _protos(protobuf_path, include_paths=None):
```

Comment:
Should we rename the function into something like `_generate_protos`? Or add docstring.